cpp/src/dictionary/decode.cu

```diff
@@ -37,15 +37,14 @@ std::unique_ptr<column> decode( dictionary_column_view const& source,
                                 rmm::mr::device_memory_resource* mr,
                                 cudaStream_t stream)
 {
-    if( source.size()==0 || source.keys_size()==0 )
+    if( source.size()==0 )
         return make_empty_column( data_type{EMPTY} );
-    auto keys = source.keys();
-    auto indices = source.indices();
-    if( indices.size()==0 )
-        return make_empty_column( keys.type() );
-
+        
+    column_view indices{ cudf::data_type{cudf::INT32}, source.size(), 
```

Comment:
This seems like a double offset.  ::data() sends in base+offset, and then you're also passing in .offset().  Should it be ::head()?

Reply:
The child columns never have offsets so `data()` and `head()` will always be the same.

Reply:
Still, seems like ::head() is semantically more correct and safer.  Would futureproof this against unexpected changes.

Reply:
Still,  head() would be semantically correct if you're also explicitly passing in offset().  If in the future, a positive offset did appear, this would quietly break.

Reply:
I think you are right. It would make it clearer what is happening. I will change both of these to use `head()` instead.